pkg/s3-proxy/server/internal-server.go

```diff
@@ -54,6 +54,15 @@ func (svr *InternalServer) GenerateServer() {
 		Addr:    addr,
 		Handler: r,
 	}
+
+	// Get the TLS configuration (if necessary)
+	tlsConfig, err := generateTLSConfig(cfg.InternalServer.SSL, svr.logger)
+	if err != nil {
+		svr.logger.Fatal(err)
```

Comment:
Can you change the function signature to have it the same as in Server? To handle that error in the main like the other server.
Moreover, why the Listen function is untouched? (Line 36)

Reply:
I'm not sure which function signature you're referring to here. (I didn't change the signature of `GenerateServer()` here, and `generateTLSConfig` is the same function from `tls.go`—can you clarify?)

I had missed the `Listen` function here (for some harebrained reason I thought it was using the implementation from `server.go`) — will do!

Reply:
No problem. I was referring to `GenerateServer()`. Can you change it `GenerateServer() error` and manage the error in the cmd/main like for server?

Reply:
Got it. Fixed both of these issues in 004e43f642e5aa047b93146b912338bfb895c423.